Kernel_23/include/CGAL/Kernel/function_objects.h

```diff
@@ -246,6 +246,8 @@ namespace CommonKernelFunctors {
       const Vector_3 abad1 = xproduct(ab1, ad1);
       const FT sc_prod_1 = abac1 * abad1;
 
+      if (abac1==NULL_VECTOR || abad1==NULL_VECTOR) return SMALLER;
```

Comment:
Why don't you do this test before the multiplication?

Reply:
more expensive